src/main/java/com/flowpowered/noise/module/modifier/Range.java

```diff
@@ -0,0 +1,122 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.flowpowered.noise.module.modifier;
+
+import com.flowpowered.noise.exception.NoModuleException;
+import com.flowpowered.noise.module.Module;
+
+/**
+ * A modifier module to map a value from one range to another.
+ * For example: 0.5 (-1 &lt; x &lt; 1) -&gt; 0.75 (0 &lt; x &lt; 1)
+ */
+public class Range extends Module {
+    
+    public static final double DEFAULT_CURRENT_LOWER_BOUND = -1f;
+    public static final double DEFAULT_CURRENT_UPPER_BOUND = 1f;
+    public static final double DEFAULT_NEW_LOWER_BOUND = 0f;
+    public static final double DEFAULT_NEW_UPPER_BOUND = 1f;
+    
+    /* Current lower bound */
+    private double currentLowerBound = DEFAULT_CURRENT_LOWER_BOUND;
+    /* Current upper bound */
+    private double currentUpperBound = DEFAULT_CURRENT_UPPER_BOUND;
+    /* New lower bound */
+    private double newLowerBound = DEFAULT_NEW_LOWER_BOUND;
+    /* New Upper Bound */
+    private double newUpperBound = DEFAULT_NEW_UPPER_BOUND;
+
+    public Range() {
+        super(1);
+    }
+
+    public double getCurrentLowerBound() {
+        return currentLowerBound;
+    }
+
+    public void setCurrentLowerBound(double currentLowerBound) {
+        this.currentLowerBound = currentLowerBound;
+    }
+
+    public double getCurrentUpperBound() {
+        return currentUpperBound;
+    }
+
+    public void setCurrentUpperBound(double currentUpperBound) {
+        this.currentUpperBound = currentUpperBound;
+    }
+
+    public double getNewLowerBound() {
+        return newLowerBound;
+    }
+
+    public void setNewLowerBound(double newLowerBound) {
+        this.newLowerBound = newLowerBound;
+    }
+
+    public double getNewUpperBound() {
+        return newUpperBound;
+    }
+
+    public void setNewUpperBound(double newUpperBound) {
+        this.newUpperBound = newUpperBound;
+    }
+    
+    /**
+     * Configure bounds for map module
+     * @param currentLower current lower bound
+     * @param currentUpper current upper bound
+     * @param newLower new lower bound
+     * @param newUpper new upper bound
+     */
+    public void setBounds(double currentLower, double currentUpper, double newLower,
+    		double newUpper) {
+    	setCurrentLowerBound(currentLower);
+    	setCurrentUpperBound(currentUpper);
+    	setNewLowerBound(newLower);
+    	setNewUpperBound(newUpper);
+    }
+
+    @Override
+    public int getSourceModuleCount() {
+        return 1;
+    }
+
+    @Override
+    public double getValue(double x, double y, double z) {
+        if (sourceModule == null) {
+            throw new NoModuleException();
+        }
+        if (sourceModule[0] == null) {
+            throw new NoModuleException();
+        }
+        
+        double oldVal = sourceModule[0].getValue(x, y, z);
+        return newLowerBound + (oldVal - currentLowerBound) * 
```

Comment:
Tried that, on paper it leads to a pretty complex equation, that will probebly won't be fun to maintain.